src/components/QuickOpenModal.js

```diff
@@ -121,33 +127,59 @@ export class QuickOpenModal extends Component<Props, State> {
     this.setState({ results });
   };
 
+  generalQuery = (query: string) => {
+    const results = formatGeneralQuery();
+    this.setState({ results });
+    return;
+  };
+
   updateResults = (query: string) => {
     if (this.isSymbolSearch()) {
       return this.searchSymbols(query);
     }
+    if (this.isGeneralQuery()) {
+      return this.generalQuery(query);
+    }
     return this.searchSources(query);
   };
 
+  updateQuery = (item: FormattedGeneralQuery) => {
+    const { setQuickOpenQuery } = this.props;
+    if (item.id == "id.123") {
+      setQuickOpenQuery("@");
+    } else if (item.id == "id.124") {
+      setQuickOpenQuery("#");
+    } else {
+      setQuickOpenQuery(":");
+    }
+  };
+
   selectResultItem = (
     e: SyntheticEvent<HTMLElement>,
-    item: ?FormattedSource | ?FormattedSymbolDeclaration
+    item:
```

Comment:
Really thinking about it this type definition shouldn't be a union and should be a single type with optional properties.